#  Лабораторна робота №2
### навчально-методичний додаток
***

## Завдання № 1 "Навігація в Jupyter Notebook"


- Продивитись __User Interface Tour__
- Відкрити [довідник команд](https://cheatography.com/weidadeyue/cheat-sheets/jupyter-notebook/pdf_bw/)
- користуючись довідником виконати операції, які наведено в наступних рядках


In [ ]:
# перетворити цей рядок в рядок-коментар та ввести своє прізвище в форматі Title-2
# Будюка

### Видалити цей рядок

### додати знизу пустий рядок

In [ ]:
# відключити в цьому рядку нумерацію строк

### Додати до цього рядка зверху 2 рядка:
1. Командний рядок
2. Рядок коментарів

## Завдання №2 " Конструкції Python"
- заповнити пусті рядки вашим рішенням

### Змінні

Ім'я, яке використовується для позначення чогось або значення, називається **змінною**. У python можна оголошувати змінні і присвоювати йому значення наступним чином:

- x = 10             
- y = -3.1428         
- s = "Вася Пупкін"
- a = [x, y, s]
- b = False
- t = (y, "Студент")

---
_[Прочитати про типи даних](http://ruslan.rv.ua/python-starter/module01/01-05_types/)_

Тип даних можна перевірити наступною інструкцією: `type(<імя змінної>)`

In [1]:
# виконайте присвоювання та виведіть типи даних x, y, s, b, t:
x = 10
y = -3.1428
s = "Вася Пупкін"
a = [x, y, s]
b = False
t = (y, "Студент")
# Виведення типів данних:
print([type(x), type(y), type(s), type(a), type(b), type(t)])

[<class 'int'>, <class 'float'>, <class 'str'>, <class 'list'>, <class 'bool'>, <class 'tuple'>]


### Опрератори

#### арифметичні

| Позначка | Виконує |
|----|---|
| +  | Додавання |
| -  | Віднімання |
| /  | поділ |
| %  | остаток від ділення |
| *  | множення |
| //  | цілочісельне ділення |
| **  | ступінь |

In [2]:
# застосуйте вищенаведені оператори до `x` та `y`:
sum = x + y
sub = x - y
div = x / y
res = x % y
mul = x * y
idv = x // y
pow = x ** y
print(f"sum = {sum}")
print(f"sub = {sub}")
print(f"div = {div}")
print(f"res = {res}")
print(f"mul = {mul}")
print(f"idv = {idv}")
print(f"pow = {pow}")

sum = 6.857200000000001
sub = 13.1428
div = -3.1818760341097114
res = -2.5711999999999993
mul = -31.427999999999997
idv = -4.0
pow = 0.0007197803728026997


#### порівняння
| Позначка | Результат |
|----|---|
| == | True, якщо рівні |
| !=  | True, якщо не рівні |
| < | менше |
| > | більше |
| <=  | менше чи равно |
| >=  | більше чи равно |

In [3]:
# застосуйте вищенаведені оператори до `x` та `y`:
eql = x == y
neq = x != y
les = x < y
mor = x > y
leq = x <= y
meq = x >= y
print(f"eql = {eql}")
print(f"neq = {neq}")
print(f"les = {les}")
print(f"mor = {mor}")
print(f"leq = {leq}")
print(f"meq = {meq}")

eql = False
neq = True
les = False
mor = True
leq = False
meq = True


### Вбудовані функції перетворення типів
1. str(<змінна чи вираз>) - перетворення в строку
2. int(<змінна чи вираз>) - перетворення в ціле
3. float(<змінна чи вираз>) - перетворення в дийсне число

[Прочитати про інші функції перетворення типів](https://pyneng.readthedocs.io/ru/latest/book/04_data_structures/convert_type.html)


In [4]:
# застосуйте (1) функцію до змінних 'x' - 't' , (2) до `x` - 's', (3) - `x` - `y`
# (якщо будуть помилки - виправте їх самотуж)
print(f"(1) str  : {[str(x), str(y), str(s), str(a), str(b), str(t)]}")
print(f"(2) int  : {[int(x), int(y)]}")
print(f"(3) float: {[float(x), float(y)]}")

(1) str  : ['10', '-3.1428', 'Вася Пупкін', "[10, -3.1428, 'Вася Пупкін']", 'False', "(-3.1428, 'Студент')"]
(2) int  : [10, -3]
(3) float: [10.0, -3.1428]


## Завдання №3 "Введення/вивід з консолі"


- __input__([символьний вираз]) - введення с консолі в _символьному форматі_ (__STDIN__) 
- __print__(<вираз>) - виведення в консоль (__STDOUT__)

Для зчитування рядка зі стандартного вводу використовується функція input (), яка зчитує рядок з клавіатури і повертає значення, яке відразу ж можна привласнити змінним:
`s = input()`

Функція `input` повертає текстовий рядок. Якщо потрібно зробити так, щоб змінні мали чисельні значення, то відразу ж після зчитування виконаємо перетворення типів за допомогою функції `int` або `float`, і запишемо нове значення:
`s1 = int(s)`

Краще об'єднати зчитування рядків і перетворення типів, якщо викликати функцію int/float для того значення, яке поверне функція input ():
`s = int(input())`

In [5]:
# ввести з консолі число 2.71 в змінну 'e'
e = float(input("Введіть e: "))

Введіть e: 2.71


In [6]:
# вивести в консоль 'е' в супені 5
e ** 5

146.1660310351

#### форматування  при виведенні
у Python для позначення рядка використовуються одинарні, подвійні та потрійні лапки. Більшість використовують одинарні лапки при оголошенні одного символу. Подвійні лапки при оголошенні рядка та потрійні лапки при оголошенні абзацу / кількох рядків.

- 'A' - одиночний символ
- "Студент Вася" - рядок символів
- """ Перший рядок 
    
      Другий рядок
      
      ..."""


In [7]:
# Рядки можуть бути призначені змінним наприкалд `name` та `surname`, 
# які можна викликати при використанні оператора print.
# Виконайте код, який наведено нижче
name = "Вася"
surname = "Пупкін"
print (name, surname)
print("1+1", 2, sep=" -> ")

Вася Пупкін
1+1 -> 2


Напишіть нижче, для чого нужна '`,`' та `sep` в функції `print()`:

Кома потрібна для виведення значень, а параметр 'sep' використовується для розділення значень виведення.

### форматування в функції `print()`

[Прочитати про методи форматування вивода](https://webdevblog.ru/f-stroki-python-3-uluchshennyj-sintaksis-formatirovaniya-strok/)

In [8]:
# виконати код та розібратись, чому функція `print()...format` спрацювала таким чином
print("{} {} - це студент".format(name, surname))
print("Результат буде = {}".format(42 / 5 + 4))
b = True
print("{} - {} хлопець!".format(name, b))
print("{} {}".format(surname, name))

Вася Пупкін - це студент
Результат буде = 12.4
Вася - True хлопець!
Пупкін Вася


## Завдання №4 "Керування обчисленнями"

### Умовний оператор if / if-else / if-elif

`if <логічний вираз(умова)> :
  <код>`
  
`if <логічний вираз(умова)> :
  <код>
 else :
    <код>`

`if <логічний вираз(умова)> :
  <код>
 elif <логічний вираз_1(умова)> :
  <код>
 elif <логічний вираз_2(умова)> :
  <код>
... 
 [else] :
   <код>`

In [9]:
x = float(input("Введи x: "))

# ----- 1 ----------
if x > 5:
    print (x)
    
# ----- 2 ----------    
if -5 < x < 5 :
    print (x)
else:
    print (x ** 2)
    
# ----- 3 ----------
if x > -5:
    print (x)
elif -5 < x < 5:
    print (x ** 2)
elif x > 5:
    print (x ** (1/2))

Введи x: 31.5
31.5
992.25
31.5


напишіть, який __if__ спрацював при:

x = -20     .... 2-й спрацював

x = 31.5    .... виводить помилку (якщо `int()`); усі 3 спрацювали (якщо `float()`)

x = 4       .... 2-й і 3-й спрацювали

In [10]:
# напишить нижче код, який виводить слово 'студент', 'школяр' або 'малятко' 
# в залежності від введенного через консоль віку

year = int(input("Введіть вік: "))

if 0 < year < 6:
    print(f"малятко")
elif year < 18:
    print(f"школяр")
else:
    print(f"студент")

Введіть вік: 20
студент


### Цикли

#### for

`for <змінна> in <перераховуємий об'єкт>
    код`
    
    
__перераховуємий об'єкт__ - це об'єкт, який складається із декількох послідовних складових (чисел, строк та ін.)

- 1, 13, -10, 321
- "Вася", "Пупкін", "студент"
- "Вася", -10.1, True

[Прочитати про функцію `range`](https://python-scripts.com/range)

In [11]:
# приклад цикла з 'for' та 'if'
for i in range(10):
    if i % 2 != 0:
        print (-i)
    else:
        pass

-1
-3
-5
-7
-9


напишить своїми словами, що виконує наведений вище код

Вище наведений код виконує послідовний циклічний прохід по цілим значенням від 0 до 10 (не включно) з кроком 1, й виводить від'ємні значення тих чисел, якщо їх остача при діленні на 2 не рівна 0 (нулю).

In [12]:
# напишіть нижче код, який сумує числа, які діляться без остатка на 3 
# в інтервалі від 10 до 100

sum = 0

for i in range(10, 101):
    if i % 3 == 0:
        sum += i
    else:
        pass
print(f"sum = {sum}")

sum = 1665


#### while

[прочитати про цикл while](https://pythontutor.ru/lessons/while/)

In [13]:
# запустіть цей код 
while(input("тисни тут: ") != '0'):
    print ("було введено")

тисни тут: 
було введено
тисни тут: 1
було введено
тисни тут: 0


поясніть своїми словами, що виконує наведений вище код

Наведений вище код виконує циклічне виведення запису "було введено", якщо введене значення не рівне 0 (нулю).

In [14]:
# використовуючи 'while`напишіть нижче код, який сумує числа, 
# які діляться без залишка на 3 в інтервалі від 10 до 100

i = 10
sum = 0

while (i < 101):
    if i % 3 == 0:
        sum += i
    i+=1
print(f"sum = {sum}")

sum = 1665
